### PR TITLE
Force SDL_WindowFlags enum to use 64bit values

### PR DESCRIPTION
**Pull request by LiquidityC**
`enum SDL_WindowFlags` is running out of space for additional flags (32bit). This change will coerce the compiler into using 64 bit values for the enum instead.

According to C standard, enums should default to the value type `int`, if any value assigned to an an enum through the `=` sign that is *larger* then `int` the compiler should use a larger size type to incorporate its largest values.

*We think this should work the same for most/all compilers*

## Description

- Tacked on `ULL` to every value in the SDL_WindowFlags declaration.
- Changed `SDL_WindowFlag` argument type to `Uint64` in `SDL_CreateWindow`
- Changed return type of `SDL_GetWindowFlags` to `Uint64`
- Update type of variables that are impacted by the `SDL_GetWindowFlags` and `SDL_CreateWindow` changes.

## Existing Issue(s)
Fixes: #7321

### CODE REVIEW

File: test/testautomation_video.c [hunk 1]
```diff
@@ -324,7 +324,7 @@ int video_getWindowFlags(void *arg)
     if (window != NULL) {
         actualFlags = SDL_GetWindowFlags(window);
         SDLTest_AssertPass("Call to SDL_GetWindowFlags()");
-        SDLTest_AssertCheck((flags & actualFlags) == flags, "Verify returned value has flags %d set, got: %" SDL_PRIu32, flags, actualFlags);
+        SDLTest_AssertCheck((flags & actualFlags) == flags, "Verify returned value has flags %u set, got: %" SDL_PRIu64, flags, actualFlags);
```
Comment (madebr) [inline test/testautomation_video.c:327]: `SDL_WindowFlags flags;` is also 64-bit.

Comment (LiquidityC) [reply test/testautomation_video.c]: Use `SDL_PRIu64` for both? Done.

Comment (LiquidityC) [reply test/testautomation_video.c:327]: Didn't compile for a lot of systems. I'll switch it back to `%u`.

Question is if some compilers are "smart" enough to realize that we aren't using any bits above 32 yet and just respecting the unsigned appendix on the enum? I'm not sure. Need to read up on standard.

**Comment by meyraud705** [comment]
Adding `ULL` to values does not make the enum 64 bit: `1ULL` is representable with 32 bit as `1U`. You need to add a dummy value bigger than `UINT32_MAX`.

You can use SDL_COMPILE_TIME_ASSERT() to check the size .

**Comment by LiquidityC** [comment]
> Adding `ULL` to values does not make the enum 64 bit: `1ULL` is representable with 32 bit as `1U`. You need to add a dummy value bigger than `UINT32_MAX`.
> 
> You can use SDL_COMPILE_TIME_ASSERT() to check the size .

Figured. Any suggestion on that dummy value? How are they usually introduced/named?

File: include/SDL3/SDL_video.h [hunk 2]
```diff
@@ -131,8 +131,14 @@ typedef enum
     SDL_WINDOW_VULKAN               = 0x10000000,   /**< window usable for Vulkan surface */
     SDL_WINDOW_METAL                = 0x20000000,   /**< window usable for Metal view */
 
+    /* This value is added to ensure SDL_WindowFlags size is compiled to 8 bytes (Uint64) */
+    __SDL_WINDOWFLAGS_SENTINEL      = UINT64_MAX,   /**< dummy value. Not for use in app code */
```
Comment (sezero) [inline include/SDL3/SDL_video.h:135]: `SDL_MAX_UINT64`

Comment (LiquidityC) [reply include/SDL3/SDL_video.h]: Done.

**Comment by eloj** [comment]
See also "[N3029: Improved Normal Enumerations](https://open-std.org/JTC1/SC22/WG14/www/docs/n3029.htm)" and "[N3030
Enhancements to Enumerations](https://open-std.org/JTC1/SC22/WG14/www/docs/n3030.htm)"

**Comment by eloj** [comment]
> Those are targeted at C23. We're not there yet, right?

No. I simply posted those because I think they provide good background information on the issue(s).

E.g that this 'trick' relies on a very common compiler extension, but it's not mandated by any pre-C23 standard. As such, it'll raise a warning when compiled with `-pedantic` on both gcc and clang.

**Comment by sezero** [comment]
> This one is giving me trouble. I'll figure it out but if anyone knows an obvious solution please share. My windows computer is out of service so I'm not able to wrangle this locally at the moment.
> 
> image
> 
> This is the line it doesn't like: image

Reproduced with VS2022 using the MSVC project. Don't know why is that happening.
PCH maybe?? (doesn't make sense..)

**Comment by sezero** [comment]
The mac CI builds are failing because of these lines in SDL_stdinc.h:
```c
#elif defined(__LP64__) && !defined(__APPLE__)
#define SDL_PRIu64 "lu"
#else
#define SDL_PRIu64 "llu"
```
@slouken: Why are we actually excluding `__APPLE__` from `__LP64__` ?

**Comment by sezero** [comment]
> MSVC is not conforming to standard by default: https://learn.microsoft.com/en-us/cpp/build/reference/zc-enumtypes?view=msvc-170

Ugh.. If we actually use that `/Zc:enumTypes`, other normal enums will most certainly break..

**Comment by LiquidityC** [comment]
Should we maybe just drop the assertion? We still have the print formatting which will cause compilation/CI issues if the size is mistakenly undone.

**Comment by sezero** [comment]
> Should we maybe just drop the assertion? We still have the print formatting
> which will cause compilation/CI issues if the size is mistakenly undone.

Possibly. And avoid using SDL_WindowFlags as a variable type and always
use Uint64 instead.

Also, we can always go back to plain ULL macros instead of an enum, which I
think would be foolproof.

**Comment by LiquidityC** [comment]
> Also, we can always go back to plain ULL macros instead of an enum, which I
> 
> think would be foolproof.

Personally I favor macros for these scenarios. But perhaps that should be done in its own PR if we want to go that way. It will require some documentation updates across a lot of files.

**Pull request closed without merging**